### ISSUE TITLE
onBlur behaviour is affected by "touch capabilities"

### ISSUE THREAD

**Issue by theKashey** [JedWatson/react-select#3737]
# Are you reporting a bug or runtime error?

That's not really a bug, and not an error - this is not correct behavior.

There is an option - [blurInputOnSelect](https://github.com/JedWatson/react-select/blob/97fbd4c3a24952ab103ca197d912f37c6057674e/packages/react-select/src/Select.js#L250)

- `blurInputOnSelect: isTouchCapable(),`

and the effect of it quite [simple](https://github.com/JedWatson/react-select/blob/97fbd4c3a24952ab103ca197d912f37c6057674e/packages/react-select/src/Select.js#L665-L667):

```js
  selectOption = (newValue: OptionType) => {
    ....
    if (blurInputOnSelect) {
      this.blurInput();
    }
```

## Problems

1. __Should it be delayed?__ In normal conditions there are two different events (click/change, blue) comes one after another with a little "gap" between them. And there is no gap here.

2. __Should it be?__ For all "touch devices", windows 10 on hybrid devices included, or only when select is caused by "touch" event? Probably [here](https://github.com/JedWatson/react-select/blob/97fbd4c3a24952ab103ca197d912f37c6057674e/packages/react-select/src/Select.js#L1298) we should check `touch event`, and execute `blur` only there, but not as a part of `selectOption`

3. Is it a real problem? Yes, it is. We run into it twice, SEV1s both times.

**Comment by tstelzer** [comment]
I believe we are having issues stemming from this as well, on touch devices the `onBlur` triggers before `onChange`, sometimes, which prevents the change event entirely.

**Comment by theKashey** [comment]
In our case - `onBlur`, fired with the still old event, was canceling `onChange`.
Just wrapping onBlur with `setImmediate` fixed the problem, and that's a quick fix we might do. Adding blur only for touch-driven selects would make it even better, but that's something good to have, not crusial to have.

**Comment by tstelzer** [comment]
Yup, though we're using `setTimeout(f, 0)` instead of `setImmediate(f)`.

**Comment by theKashey** [comment]
Who will try `Promise.resolve().then(onBlur)`? Not sure that would not be too fast.
`setImmediate` is more "correct" way to emulate "browser tick", however, has to be polyfilled.

**Comment by GeorgeMarcus** [comment]
I wasted an entire day on this. I couldn't figure out why my test wasn't working. Apparently the test environment has touch capabilities...

**Comment by theKashey** [comment]
Lucky one, caught it in tests :)

**Comment by karlvr** [comment]
We appear to have this issue with `Creatable` on iOS (iPad only) where the `<input>` is receiving the blur event before the chosen option receives its click event, and so the menu closes before an option can be chosen! I'm currently trying to create a workaround. This used to work, so I'm not sure if it's an iOS behaviour change.
